tests/AbstractFactoryDecoratorTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace byrokrat\id;
+
+use PHPUnit\Framework\TestCase;
+
+class AbstractFactoryDecoratorTest extends TestCase
+{
+    public function testConstruct()
+    {
+        $factory = new class extends AbstractFactoryDecorator {
+            public function createNewInstance(?string $raw): IdInterface {}
+        };
+
+        $this->assertInstanceOf(IdFactory::class, $factory);
+    }
+
+    public function testCreateId()
+    {
+        $factory = new class extends AbstractFactoryDecorator {
+            public function createNewInstance(?string $raw): IdInterface {
+                return new class extends AbstractId {};
+            }
+        };
+
+        $this->assertInstanceOf(AbstractId::class, $factory->createId('123'));
+    }
+
+    public function testUseNextFactoryIfIdCannotBeCreated()
+    {
+        $nextFactory = $this->createMock(AbstractFactoryDecorator::class);
+        $nextFactory->expects($this->once())->method('createId');
```

Comment:
Return something so that we can assert later?

Reply:
We test that the next factory is called. What it returns is not important.